editoast/src/models/paced_train.rs

```diff
@@ -71,6 +71,19 @@ impl PacedTrain {
             options: self.options,
         }
     }
+
+    pub fn iter_occurrences(&self) -> impl Iterator<Item = TrainSchedule> + use<'_> {
```

Comment:
nitpick: I actually find weird in terms of API that `into_first_occurrence`, which yields a `TrainSchedule`, takes ownership of `self`, but `iter_occurences`, which also yields instances of `TrainSchedule`, takes only a reference of `self`. Maybe this function should also take ownership and let the caller decide if a `.clone()` is needed. What do you think?

note: might need to be renamed `into_iter_occurrences` if we choose to do this.

Reply:
Ideally, the signature should look like this imo:

```rust
pub fn iter_occurrences(&self) -> impl Iterator<Item = &'self TrainSchedule>
```

The problem is that every `train_simulation_batch` functions and friends take a `&[TrainSchedule]` and not a `&[&TrainSchedule]` (as they should since they do not consume the schedule), making the iterator item useless.

So since we want a `Item = TrainSchedule`, we still need to clone the base schedule once per occurrence. My point being, since we need to clone anyway, something we can't hide from the caller, let's clone once more for the `into_first_occurrence` and not consume `self`, bringing us closer to a more sound API.

Wdyt? @woshilapin @younesschrifi (unresolving)

Reply:
@younesschrifi let's give @woshilapin the chance to agree or disagree before resolving and merge the pr 😄